### ISSUE TITLE
Some functions don't do anything if not explicitly configured (For Enlightenment Desktop Environment)

### ISSUE THREAD

**Issue by hupfdule** [Chris2011/QuickOpener-NetBeans#5]
Software | Version
----|----
QuickOpener Plugin: | 1.4.0
Netbeans: | 11.0
Operating System: | Debian Buster
Desktop Environmen: | Enlightenment E22 (0.22.4)

"Open in Terminal" and "Open in File Manager" must be configured via "Tools / Options / Miscellaneous / QuickOpener / General Options" to have any effect.

If I don't do this, nothing happens. No error message. Not even on stdout or stderr. In `$HOME/.netbeans/11.0/var/log/uigestures` the following entry can be found, but nothing else:

```
<record>                                                                                                                                                              
  <date>2019-07-11T16:52:42</date>                                                                                                                                    
  <millis>1562856762555</millis>                                                                                                                                      
  <sequence>5668</sequence>                                                                                                                                           
  <logger>org.netbeans.ui.actions</logger>                                                                                                                            
  <level>400</level>                                                                                                                                                  
  <thread>39</thread>
  <message>UI_ACTION_BUTTON_PRESS</message>
  <key>UI_ACTION_BUTTON_PRESS</key>
  <catalog>org.openide.awt.Bundle</catalog>
  <param>javax.swing.JButton[]</param>
  <param>javax.swing.JButton</param>
  <param>me.dsnet.quickopener.actions.Terminal[Open in Terminal]</param>
  <param>me.dsnet.quickopener.actions.Terminal</param>
  <param>Open in Terminal</param>
</record>
```

**Comment by Chris2011** [comment]
> "Open in Terminal" and "Open in File Manager" must be configured via "Tools / Options / Miscellaneous / QuickOpener / General Options".
A clear error message should be displayed in case nothing is configured yet.
It would also be nice if there were sensible defaults already filled in.

First no you don't need to configure it. If you just install. Open in Terminal, Open in File Manager, copy path, open in default app, should work out of the box.

What you can do is, to have a look into the IDE log, when you hit a shortcut or click the buttons in the toolbar. And what you also can do is to use a fresh NetBeans and try it there. If it is working there, there must be a conflict with another plugin.

>Sorry that this issue has "evolved" a bit. It now contains some different topics, but as I still see them related, I leave them in this single report. If you want I can separate them into several issues.

Yes you are right with evolved and those are of course different topics. The topic here is that most of the functions are not working, what you said. So there is no dependent behaviour to the apply buttons, no dependent behaviour to the mate-terminal stuff etc.

So please change the ticket. It should only contains your bug, and some info. Nothing more. Everything else, like changing behaviour of the options (multiple apply buttons to single one) is a feature request for example and should handled separately. The other stuff the same. Thx.

**Comment by hupfdule** [comment]
I have split and updated this ticket as you requested.

**Comment by Chris2011** [comment]
Thx, please add your environment to all of the other tickets, because I will not remember due to this ticket.
And to ping it up, please have a look into your IDE log, when you hit the shortcut/button. What you added, is not the ide log. the IDE log is messages.log or you can open it right inside your IDE via View -> IDE Log.

**Comment by Chris2011** [comment]
Will have a look, but again, those functions "Open in Terminal" and "Open in File manager" don't need to be configured. I don't have them configured too on my system (Windows) so there must be a different problem.

**Comment by hupfdule** [comment]
> Thx, please add your environment to all of the other tickets, because I will not remember due to this ticket.

I have put that information into the tickets where I see a releation to the desktop environment.

> And to ping it up, please have a look into your IDE log, when you hit the shortcut/button. What you added, is not the ide log. the IDE log is messages.log or you can open it right inside your IDE via View -> IDE Log.

I have done so. The messages.log ist also contained in $HOME/.netbeans/11.0/var/log. But nothing is written to that file.

**Comment by hupfdule** [comment]
> Will have a look, but again, those functions "Open in Terminal" and "Open in File manager" don't need to be configured. I don't have them configured too on my system (Windows) so there must be a different problem.

Yes, I know. Therefore I have reworded this ticket to state that difference.
This seems to be the case for an unknown desktop environment. When QuickOpener provides a configuration for my desktop environment (Enlightenment) as with my PR #13 I don't have to configure anything explicitly.

And I think there is the problem. If you look at https://github.com/Chris2011/QuickOpener-NetBeans/blob/299265450f923185a7a2887c0e7534a6f9a81191/src/main/java/com/sessonad/oscommands/detector/OperatingSystem.java#L37-L38 you can see that the terminal and file manager are set to `null`. That means configuration is necessary in that case. That should trigger and error message (and possibly open the corresponding configuration dialog).

**Comment by Chris2011** [comment]
I had a quick look into the PR #13 so this will fix this problem, right?

**Comment by Chris2011** [comment]
So the problem is about an unknown linux desktop environment "Enlightenment" what you added with your PR. That's great. Thx :)

**Comment by hupfdule** [comment]
@Chris2011 No, it doesn't. The PR is only an enhancement, since it adds support for one specific desktop environment.

The problem still exists if someone uses, for example, fvwm.

Also I don't know what happens if those default applications not installed (which may be the case). I have to try that out.

**Comment by Chris2011** [comment]
Yeah, so maybe we need to add each desktop evironment as you can see it in the list. Maybe it is, as it is.

**Comment by Chris2011** [comment]
Will merge it later. Thx :)

**Comment by hupfdule** [comment]
> Yeah, so maybe we need to add each desktop evironment as you can see it in the list. Maybe it is, as it is.

No it is not. You will never get "each existing desktop environment". And most of them (being only window managers) don't have any "default" terminal or file manager. 

The approach of the original author to have some fallback for unknown operating systems and environments or in case they couldn't be detected correctly (the detection is quite fragile) is correct.

It should just provide a good behaviour in that case. Just doing nothing isn't helpful.

**Comment by Chris2011** [comment]
Ok, so first I will close this issue here, because I think your PR fixes your specific environment. I can understand your point, for this message like a notification I can think of, you can create another ticket to handle that case if you didn't do it yet. Thx.

**Comment by hupfdule** [comment]
I still think that this would have been the correct report, since I never expected the Enlightenment would be one of the predefined Desktop Environments and this problem stays the same for all other window managers or if the detection didn't work.

However, I have done as you wanted: #15

**Comment by Chris2011** [comment]
Yeah but the title like "Some functions doesn't do anything if not....." is not correct at all. It depends on the system, which is not supported. So the correct thing is, yes it doesn't do anything, if the system is not supported, which you created another ticket to point it out, it would be better to have a notification etc.

Anyway, thx for the PR, I merged it and will release it soon :)

**Comment by Chris2011** [comment]
FIxed in 1.5.0. Thx :)
